### PR TITLE
Mob 557

### PR DESCRIPTION
**Pull request by NarendraPrabhu**
This PR contains changes involving removal of unwanted data model classes like LiString, LiBoolean, LiInt, LiInteger, LiFloat and LiLong, which were a hindrance in using with other compatible data processing libraries like gson, data-binding...etc. 
Also respective test cases are updated.

### CODE REVIEW

File: lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java [hunk 1]
```diff
@@ -85,36 +67,25 @@ public Long getWidth() {
         return width;
     }
 
-    public void setWidth(LiBaseModelImpl.LiInteger result) {
-        this.width = result.getValue();
+    public void setWidth(Long result) {
```
Comment (tanmayyy) [inline lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: Any reasons for moving from Int to Long?

Comment (tanmayyy) [inline lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: I would suggest we stick to primitive long or Long.. auto boxing is expensive

Comment (NarendraPrabhu) [reply lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: Although the name is Integer, it was holding a long value internally.

Comment (NarendraPrabhu) [reply lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: agree.

**Comment by NarendraPrabhu** [comment]
LiBaseModelImpl is a generic class used to represent the data part of all the API responses.
as in ..
`{
	"status": "success",
	"message": "",
	"http_code": 200,
	"data": { ...}
}`

File: lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java (hunk 1 above)
Comment (NarendraPrabhu) [reply lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: There are too many such all over the code, I think we can live with it.

File: lia-core/src/main/java/com/lithium/community/android/model/helpers/LiRanking.java [hunk 2]
```diff
@@ -130,32 +130,20 @@ public void setHref(String href) {
...
+        this.name = name;
     }
 
     public Long getId() {
         return id;
     }
 
-    public void setId(LiBaseModelImpl.LiInt id) {
-        this.id = id.getValue();
+    public void setId(long id) {
```
Comment (tanmayyy) [inline lia-core/src/main/java/com/lithium/community/android/model/helpers/LiRanking.java]: LiInt was earlier using Long.. I guess we can stick to that

File: lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java (hunk 1 above)
Comment (tanmayyy) [reply lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: LiInt and LiLong were earlier using Long.. I guess we can stick to that.. no point introducing primitives now

Comment (NarendraPrabhu) [reply lia-core/src/main/java/com/lithium/community/android/model/helpers/LiImage.java]: sure

File: lia-core/src/main/java/com/lithium/community/android/model/response/LiUser.java [hunk 3]
```diff
@@ -253,28 +220,16 @@ public Float getAverageMessageRating() {
...
 
-    public void setAverageMessageRating(LiBaseModelImpl.LiFloat averageMessageRating) {
-        this.averageMessageRating = averageMessageRating.getValue();
-    }
-
-    public LiBaseModelImpl.LiFloat getAverageMessageRatingAsLiFloat() {
-        final LiBaseModelImpl.LiFloat ret = new LiBaseModelImpl.LiFloat();
-        ret.setValue(getAverageMessageRating());
-        return ret;
+    public void setAverageMessageRating(float averageMessageRating) {
```
Comment (tanmayyy) [inline lia-core/src/main/java/com/lithium/community/android/model/response/LiUser.java]: we can use Float

Comment (NarendraPrabhu) [reply lia-core/src/main/java/com/lithium/community/android/model/response/LiUser.java]: done

**Pull request merged**